### DISCUSSION TITLE
Adding catalan translation - Does it behave well with a partial translation?

### DISCUSSION THREAD

**Discussion by xmontero** [benkeen/generatedata#772] [General]
In relation to https://github.com/benkeen/generatedata/discussions/770

In `client/src/i18n` I see the "main translations".

I plan to add a new language: "catalan"

* code: `ca`
* https://en.wikipedia.org/wiki/Catalan_language

Questions:

* Do I need to make "the whole translation in a shot"?
* I mean: I'd like to progressively add rows (maybe 5 per day), can't do it in a row. Does it behave well if I set "only a few keys"?

I'd be expenting that a key found in `ca.json` is used but if not found, it takes the default, so I can translate progressively. Works like that?

**Comment by benkeen** [comment]
Using "ca" is great - that's the ISO 639-1 2-char language code. 

> Do I need to make "the whole translation in a shot"? I mean: I'd like to progressively add rows (maybe 5 per day), can't do it in a row. Does it behave well if I set "only a few keys"?
> I'd be expenting that a key found in ca.json is used but if not found, it takes the default, so I can translate progressively. 

That would be nice! But at the moment, no - that's not hooked up. To be honest, the translations are pretty basic right now - there were other priorities for the script, so getting things translated is a pretty manual process.

What I'd suggest doing it making a copy of the `client/src/i18n/en.json` file and call it to `ca.json`, then you can translate the contents at your leisure. 

Similarly, all the various plugins will need a separate file too...

If you like, I could add a grunt command to at least do this for you? i.e. create copies of the English file with the new filename in all folders. Otherwise it'll be kind of a pain. Just let me know. 

Oh - and lastly, you'll need to add the new `ca` value to GD_LOCALES in `.env.default`. 

> GD_LOCALES=ar,de,en,es,fr,hi,ja,nl,pt,ta,zh

That'll then be picked up by the build and be recognized in the interface.

**Reply by xmontero** [reply]
Thanks for the grunt offer. I'll go with it by myself, no worries. Multi-cursor IDEs are fantastic for those tasks. No worries.
